staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/custom_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package modes_test
+
+import (
+	"encoding"
+	"encoding/json"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+type TextUnmarshalerPointer struct{}
+
+func (*TextUnmarshalerPointer) UnmarshalText([]byte) error { return nil }
+
+type Struct[F any] struct {
+	Field F
+}
+
+type StructCBORMarshalerValue[F any] struct {
+	Field F
+}
+
+func (StructCBORMarshalerValue[F]) MarshalCBOR() ([]byte, error) { return nil, nil }
+
+type StructCBORMarshalerPointer[F any] struct {
+	Field F
+}
+
+func (*StructCBORMarshalerPointer[F]) MarshalCBOR() ([]byte, error) { return nil, nil }
+
+type StructCBORUnmarshalerPointer[F any] struct {
+	Field F
+}
+
+func (*StructCBORUnmarshalerPointer[F]) UnmarshalCBOR([]byte) error { return nil }
+
+func TestCheckUnsupportedMarshalers(t *testing.T) {
+	t.Run("accepted", func(t *testing.T) {
+		for _, v := range []interface{}{
+			nil,
+			false,
+			0,
+			0.0,
+			"",
+			[]interface{}{nil, false, 0, 0.0, "", []interface{}{}, map[string]interface{}{}},
+			map[string]interface{}{
+				"nil":                      nil,
+				"false":                    false,
+				"0":                        0,
+				"0.0":                      0.0,
+				`""`:                       "",
+				"[]interface{}{}":          []interface{}{},
+				"map[string]interface{}{}": map[string]interface{}{},
+			},
+			[0]json.Marshaler{},
+			[3]bool{},
+			[1]interface{}{},
+			Struct[string]{},
+			StructCBORMarshalerValue[json.Marshaler]{},
+			&StructCBORMarshalerPointer[json.Marshaler]{},
+			map[int]interface{}{0: nil},
+			new(string),
+			map[cbor.Marshaler]cbor.Marshaler{},
+			make([]string, 10),
+			[]Struct[interface{}]{{Field: true}},
+			&Struct[StructCBORUnmarshalerPointer[json.Unmarshaler]]{},
+			map[encoding.TextMarshaler]struct{}{},
+			map[string]json.Marshaler{},
+		} {
+			if err := modes.RejectCustomMarshalers(v); err != nil {
+				t.Errorf("%#v: unexpected non-nil error: %v", v, err)
+			}
+		}
+	})
+
+	t.Run("rejected", func(t *testing.T) {
+		for _, v := range []interface{}{
```

Comment:
Is the embedded type case in here? I'm failing to eyeball it.

Reply:
Added embedded cases. I saw them being exercised in things like `metav1.Time` and had missed them here.